flink-connectors/flink-connector-kinesis/src/main/java/org/apache/flink/streaming/connectors/kinesis/internals/publisher/fanout/FanOutRecordPublisherConfiguration.java

```diff
@@ -0,0 +1,438 @@
+	private final long deregisterStreamBaseBackoffMillis;
+
+	/**
+	 * Maximum backoff millis for the deregister stream operation.
+	 */
+	private final long deregisterStreamMaxBackoffMillis;
+
+	/**
+	 * Exponential backoff power constant for the deregister stream operation.
+	 */
+	private final double deregisterStreamExpConstant;
+
+	/**
+	 * Maximum retry attempts for the deregister stream operation.
+	 */
+	private final int deregisterStreamMaxRetries;
+
+	/**
+	 * Max retries for the describe stream operation.
+	 */
+	private final int describeStreamMaxRetries;
+
+	/**
+	 * Backoff millis for the describe stream operation.
+	 */
+	private final long describeStreamBaseBackoffMillis;
+
+	/**
+	 *  Maximum backoff millis for the describe stream operation.
+	 */
+	private final long describeStreamMaxBackoffMillis;
+
+	/**
+	 * Exponential backoff power constant for the describe stream operation.
+	 */
+	private final double describeStreamExpConstant;
+
+	/**
+	 * Max retries for the describe stream consumer operation.
+	 */
+	private final int describeStreamConsumerMaxRetries;
+
+	/**
+	 * Backoff millis for the describe stream consumer operation.
+	 */
+	private final long describeStreamConsumerBaseBackoffMillis;
+
+	/**
+	 *  Maximum backoff millis for the describe stream consumer operation.
+	 */
+	private final long describeStreamConsumerMaxBackoffMillis;
+
+	/**
+	 * Exponential backoff power constant for the describe stream consumer operation.
+	 */
+	private final double describeStreamConsumerExpConstant;
+
+	/**
+	 * Creates a FanOutProperties.
+	 *
+	 * @param configProps the configuration properties from config file.
+	 * @param streams     the streams which is sent to match the EFO consumer arn if the EFO registration mode is set to `NONE`.
+	 */
+	public FanOutRecordPublisherConfiguration(Properties configProps, List<String> streams) {
+		Preconditions.checkArgument(configProps.getProperty(ConsumerConfigConstants.RECORD_PUBLISHER_TYPE).equals(RecordPublisherType.EFO.toString()), "Only efo record publisher can register a FanOutProperties.");
+		KinesisConfigUtil.validateEfoConfiguration(configProps, streams);
+
+		efoRegistrationType = EFORegistrationType.valueOf(configProps.getProperty(ConsumerConfigConstants.EFO_REGISTRATION_TYPE, EFORegistrationType.EAGER.toString()));
+		//if efo registration type is EAGER|LAZY, then user should explicitly provide a consumer name for each stream.
+		if (efoRegistrationType == EFORegistrationType.EAGER || efoRegistrationType == EFORegistrationType.LAZY) {
+			consumerName = configProps.getProperty(ConsumerConfigConstants.EFO_CONSUMER_NAME);
+		} else {
+			//else users should explicitly provide consumer arns.
+			streamConsumerArns = new HashMap<>();
+			for (String stream : streams) {
+				String key = ConsumerConfigConstants.EFO_CONSUMER_ARN_PREFIX + "." + stream;
+				streamConsumerArns.put(stream, configProps.getProperty(key));
+			}
+		}
+
+		this.subscribeToShardMaxRetries = Integer.parseInt(
```

Comment:
@xiaolong-sn can you please refactor these initialisations to not convert the default:
- `long > String > long`

You could use optional:
```
this.subscribeToShardMaxRetries = Optional
    .ofNullable(configProps.getProperty(SUBSCRIBE_TO_SHARD_RETRIES))
    .map(Long::parseLong)
    .orElse(DEFAULT_SUBSCRIBE_TO_SHARD_RETRIES);
```

Reply:
@xiaolong-sn also there are no happy path unit tests. Please add tests that verify the configuration is parsed correctly from the properties

Reply:
Done.